apis/core/v1beta1/openstackcontrolplane_webhook.go

```diff
@@ -920,6 +920,10 @@ func (r *OpenStackControlPlane) DefaultServices() {
 			r.Spec.Horizon.Template = &horizonv1.HorizonSpecCore{}
 		}
 
+		if r.Spec.Horizon.Template.Secret == "" {
```

Comment:
isn't this [1] sufficient to inject this input?

[1] https://github.com/openstack-k8s-operators/openstack-operator/blob/main/pkg/openstack/horizon.go#L156

Reply:
Ah, yeah, it is indeed.

Ok, so we only need the webhook validation then.